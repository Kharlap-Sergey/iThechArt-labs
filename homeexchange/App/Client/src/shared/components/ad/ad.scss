@import "shared/variables.scss";
.ad {
  width: fit-content;
  min-width: 450px;
  max-width: 98%;
  padding-top: 20px;
  padding-right: 40px;
  padding-left: 40px;
  overflow: hidden;
  font-size: 1rem;
  background: linear-gradient(to right, $purple-blue-light, $green);
  border: 0;
  border-radius: 10px;
  &__header {
    display: flex;
    justify-content: space-between;
  }
  &__main {
    position: relative;
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    padding-top: 20px;
    padding-bottom: 10px;
    border-radius: 20px;
  }
  &__main-info {
    display: flex;
    justify-content: space-between;
    min-height: 65px;
  }
  &__author {
    width: 60px;
    height: 60px;
  }
  &__date {
    margin-top: 0;
    padding-top: 0;
    color: white;
  }
  &__type {
    color: $dark-blue;
    font-weight: 600;
  }
  &__title {
    margin-bottom: 0;
    color: white;
    white-space: normal;
    word-wrap: break-word;
  }
  &__discription {
    margin-top: 5px;
    color: black;
    font-size: 1.4rem;
    white-space: normal;
    word-wrap: break-word;
  }
  &__less {
    display: flex;
    justify-content: flex-end;
    &-btn {
      position: relative;
      box-sizing: border-box;
      margin-top: 20px;
      padding: 2px 8px 6px 8px;
      color: $more-btn-color;
      font-weight: 600;
      font-size: 2rem;
      text-align: center;
      background-color: $dark-blue;
      border: 0;
      border-radius: 12px;
      outline: none !important;
      cursor: pointer;
      -webkit-transition: all 0.5s 0s ease-in-out;
      -moz-transition: all 0.5s 0s ease-in-out;
      -ms-transition: all 0.5s 0s ease-in-out;
      -o-transition: all 0.5s 0s ease-in-out;
      transition: all 0.5s 0s ease-in-out;
      &:active {
        background-color: $dark-green;
      }
      &:hover {
        z-index: 2;
        color: $dark-blue;
        font-size: 1.5rem;
        background-color: $green;
        box-shadow: 0 0 100px 650px $green;
      }
    }
  }
  &__action {
    position: absolute;
    bottom: 10px;
  }
  &__action-btn {
    position: relative;
    box-sizing: border-box;
    margin-top: 20px;
    margin-left: 5px;
    padding: 2px 8px 6px 8px;
    color: $more-btn-color;
    font-weight: 600;
    font-size: 1rem;
    text-align: center;
    background-color: $dark-blue;
    border: 0;
    border-radius: 12px;
    outline: none !important;
    cursor: pointer;
    -webkit-transition: all 0.5s 0s ease-in-out;
    -moz-transition: all 0.5s 0s ease-in-out;
    -ms-transition: all 0.5s 0s ease-in-out;
    -o-transition: all 0.5s 0s ease-in-out;
    transition: all 0.5s 0s ease-in-out;
    &:active {
      background-color: $dark-green;
    }
    &:hover {
      z-index: 2;
      color: $dark-blue;
      font-size: 1.5rem;
      background-color: $green;
      box-shadow: 0 0 500px 650px $green;
    }
  }
}
