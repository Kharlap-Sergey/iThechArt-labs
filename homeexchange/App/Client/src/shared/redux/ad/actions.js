import { SET_ADS } from './constants';

export function setAdsAction(ads) {
  return {
    type: SET_ADS,
    payload: ads,
  };
}

export function clearAdsAction() {
  return {
    type: AD_CLEAR,
  };
}

export function setAdAction(ad) {
  return {
    type: AD_GET,
    payload: ad,
  };
}