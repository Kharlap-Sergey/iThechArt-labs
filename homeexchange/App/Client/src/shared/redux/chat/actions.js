import {  ADD_MESSAGES, ADD_CHATS, CLEAR } from "./constants";
export function addChatMessagesAction(messages) {
  return {
    type: ADD_MESSAGES,
    payload: messages,
  };
}
export function addChatListAction(chats) {
  return {
    type: ADD_CHATS,
    payload: chats,
  };
}
export function clearChatAction() {
  return {
    type: CLEAR,
  };
}
