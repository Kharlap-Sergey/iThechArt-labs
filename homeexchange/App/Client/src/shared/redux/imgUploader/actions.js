import { CLEAR, SET } from "./constants";

export const clearProfileImgAction = () => ({ type: CLEAR });

export const setProfileImgAction = (file, profileId) => ({
  type: SET,
  payload: {
    file,
    profileId: profileId,
  },
});
