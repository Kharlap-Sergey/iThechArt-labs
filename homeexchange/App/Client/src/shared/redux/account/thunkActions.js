import { path, pathApi } from "../../utils/path";
import { auth } from "shared/utils/auth";
import { requestWrapper } from "shared/utils/requestWrapper";
import { redirectToAction } from "shared/redux/redirect/actions";
import { loginUserAction } from "./actions";
import {
  disableAllAction,
  enableLoginAction,
  enableRegistrationActin,
} from "shared/redux/loader/actions";
import { toastrNotifier } from "shared/redux/tostrNotifier";

export function loginUserPost(user) {
  return async (dispatch) => {
    dispatch(enableLoginAction());
    try {
      const url = pathApi.account.login;
      const response = await requestWrapper.post(url, user);
      if (response.ok) {
        const data = await response.json();
        auth.setToken(data.jwt);
        dispatch(
          loginUserAction({
            email: data.user.email,
            userId: data.user.id,
          })
        );
        dispatch(redirectToAction(path.home));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}

export function registrateUserPost(user) {
  return async (dispatch) => {
    try {
      dispatch(enableRegistrationActin());
      const url = pathApi.account.registrate;
      const response = await requestWrapper.post(url, user);

      if (response.ok) {
        dispatch(redirectToAction(path.login));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}

export function updateUserPost(user) {
  return async (dispatch) => {
    try {
      dispatch(enableRegistrationActin());
      const url = pathApi.account.update;
      const response = await requestWrapper.post(url, user);
      if (response.ok) {
        dispatch(redirectToAction(path.profile.to(user.id)));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
      dispatch(disableAllAction());
    }
  };
}