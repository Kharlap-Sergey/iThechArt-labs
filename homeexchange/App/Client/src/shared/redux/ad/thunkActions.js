import { requestWrapper } from "shared/utils/requestWrapper";
import { redirectToAction } from "shared/redux/redirect/actions";
import { pathApi, path } from "shared/utils/path";
import {
  enableAdFromtActin,
  disableAllAction,
  enableAdLoader,
} from "shared/redux/loader/actions";
import { toastrNotifier } from "shared/redux/tostrNotifier";
import { setAdAction, setAdsAction } from "./actions";

export function createNewAd(ad) {
  return async (dispatch) => {
    try {
      dispatch(enableAdFromtActin());
      const url = pathApi.ad.create;
      const response = await requestWrapper.post(url, ad);
      if (response.ok) {
        dispatch(redirectToAction(path.home));
      } else {
        await toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}

export function getAd(adId) {
  return async (dispatch) => {
    dispatch(enableAdFromtActin());
    try {
      const url = pathApi.ad.get(adId);
      const response = await requestWrapper.get(url);
      if (response.ok) {
        const data = await response.json();
        dispatch(setAdAction(data));
      } else {
        const data = await response.json();
        toastrNotifier.alertBadResponse(response);
        dispatch(redirectToAction(path.Profile));
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}

export function updateAd(ad) {
  return async (dispatch) => {
    dispatch(enableAdFromtActin());
    try {
      const url = pathApi.ad.update();

      const response = await requestWrapper.post(url, ad);
      if (response.ok) {
        dispatch(redirectToAction(path.profile));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}

export function deleteAd(adId) {
  return async (dispatch) => {
    const url = pathApi.ad.delete(adId);
    const response = await requestWrapper.delete(url);
    if (response.ok) {
      dispatch(redirectToAction(path.home));
    } else {
      toastrNotifier.alertBadResponse(response);
    }
  };
}

export function replyOnAd(adId) {
  return async (dispatch) => {
    const url = pathApi.ad.replyOnAd(adId);
    const response = await requestWrapper.post(url, {});
    if (response.ok) {
      dispatch(redirectToAction(path.home));
    } else {
      toastrNotifier.alertBadResponse(response);
    }
  };
}

export function getAds(page, type, searchString, authorId) {
  return async (dispatch) => {
    try {
      dispatch(enableAdLoader());
      const url = pathApi.ad.loadPage();
      const data = {
        filter: { type, authorId, searchString },
        page,
      };
      const response = await requestWrapper.post(url, data);
      if (response.ok) {
        const data = await response.json();
        dispatch(setAdsAction(data));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
      dispatch(disableAllAction());
    }
  };
}
