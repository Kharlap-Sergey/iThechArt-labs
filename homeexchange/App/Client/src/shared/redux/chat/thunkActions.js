import { pathApi } from "shared/utils/path";
import { requestWrapper } from "shared/utils/requestWrapper";
import { redirectToAction } from "shared/redux/redirect/actions";
import { path } from "shared/utils/path";
import { addChatListAction, addChatMessagesAction } from "./actions";
import { toastrNotifier } from "shared/redux/tostrNotifier";

export function loadChatList() {
  return async (dispatch) => {
    try {
      const url = pathApi.chat.loadChatList;
      const response = await requestWrapper.get(url);
      if (response.ok) {
        const data = await response.json();
        dispatch(addChatListAction(data));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    }
  };
}

export function loadPrivateRoomId(member1, member2) {
  return async (dispatch) => {
    try {
      const url = pathApi.chat.loadChatId;
      const response = await requestWrapper.post(url, { member1, member2 });
      if (response.ok) {
        const data = await response.json();
        dispatch(redirectToAction(path.chat(data)));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    }
  };
}

export function loadChatMessages(chatId) {
  return async (dispatch) => {
    try {
      const url = pathApi.chat.loadChatMessages + `/${chatId}`;
      const response = await requestWrapper.get(url);
      if (response.ok) {
        const data = await response.json();
        dispatch(addChatMessagesAction(data));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    }
  };
}
