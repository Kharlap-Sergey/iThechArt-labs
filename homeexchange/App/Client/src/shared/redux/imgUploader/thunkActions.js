import { pathApi } from "shared/utils/path";
import { requestWrapper } from "shared/utils/requestWrapper";
import { toastrNotifier } from 'shared/redux/tostrNotifier';
import { setProfileImgAction } from "./actions";

export function sendFile(file, authorId) {
  return async (dispatch) => {
    try {
      const url = pathApi.img.send();
      const response = await requestWrapper.postFiles(url, file);
      if (response.ok) {
        dispatch(downloadFile(authorId));
      } else {
        toastrNotifier.alertBadResponse(response);
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
    }
  };
}

export function downloadFile(profileId) {
  return async (dispatch) => {
    try {
      const url = pathApi.img.get(profileId);
      const response = await requestWrapper.getFile(url);
      if (response.ok) {
        const data = await response.blob();
        var reader = new FileReader();

        reader.onload = function (event) {
          dispatch(setProfileImgAction(event.target.result, profileId));
        };

        reader.readAsDataURL(data);
      } else {
      }
    } catch (e) {
      toastrNotifier.tryAgainLater();
    } finally {
    }
  };
}
