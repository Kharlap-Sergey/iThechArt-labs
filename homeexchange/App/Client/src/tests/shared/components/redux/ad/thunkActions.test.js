import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import * as thunkActions from "shared/redux/ad/thunkActions";
import fetchMock from "fetch-mock";
import { path, pathApi } from "shared/utils/path";
import { disableAllAction, enableAdFromtActin } from "shared/redux/loader/actions";
import { redirectToAction } from "shared/redux/redirect/actions";

const middleware = [thunk];
const mockStore = configureMockStore(middleware);


describe("test thunk actions", () => {
  const mockAd = {
    id: 1,
    title: "some",
    discription: "some",
    dateOfPublication: Date.now().toString(),
    type: 1,
    userId: 1,
  };
  const initialStore = {
    ads: {
      ad: mockAd,
      adsPage: {
        hasNext: false,
        hasPrevious: false,
        ads: [],
      },
    },
  };

  afterEach(() => {
    fetchMock.restore();
  });
  it("test create new ad to bew not permited", () => {
    fetchMock.postOnce(pathApi.ad.create, {
      status: 401,
      body: { text: "some" },
      headers: { "content-type": "application/json" },
    });

    const expectActions = [
      enableAdFromtActin(),
      disableAllAction(),
    ]
    const store = mockStore(initialStore);

    store.dispatch(thunkActions.createNewAd(mockAd)).then(()=>{
      expect(store.getActions()).toBe(expectActions);
    });
  });
  
  it("test create new ad to be success", () => {
    fetchMock.postOnce(pathApi.ad.create, {
      body: { text: "some" },
      headers: { "content-type": "application/json" },
    });

    const expectActions = [
      enableAdFromtActin(),
      redirectToAction(path.home),
      disableAllAction(),
    ]
    const store = mockStore(initialStore);

    store.dispatch(thunkActions.createNewAd(mockAd)).then(()=>{
      expect(store.getActions()).toBe(expectActions);
    });
  });
});
