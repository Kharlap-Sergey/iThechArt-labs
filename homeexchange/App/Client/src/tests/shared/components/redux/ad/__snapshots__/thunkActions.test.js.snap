// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test thunk actions test create new ad to bew not permited 1`] = `
Array [
  Object {
    "type": "loader/enable/adform",
  },
  Object {
    "type": "loader/disable/all",
  },
]
`;
